apps/api/src/app/rate-limiting/usecases/get-api-rate-limit-service-maximum-config/get-api-rate-limit-service-maximum-config.usecase.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import {
+  ApiRateLimitCategoryEnum,
+  ApiRateLimitServiceMaximumEnvVarFormat,
+  ApiServiceLevelEnum,
+  DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG,
+  IApiRateLimitServiceMaximum,
+} from '@novu/shared';
+import { createHash } from 'crypto';
+import {
+  buildMaximumApiRateLimitKey,
+  buildServiceConfigApiRateLimitMaximumKey,
+  CacheService,
+  InvalidateCacheService,
+} from '@novu/application-generic';
+
+@Injectable()
+export class GetApiRateLimitServiceMaximumConfig implements OnModuleInit {
+  public default: IApiRateLimitServiceMaximum = DEFAULT_API_RATE_LIMIT_SERVICE_MAXIMUM_CONFIG;
+
+  constructor(private invalidateCache: InvalidateCacheService, private cacheService: CacheService) {}
+
+  async onModuleInit() {
+    await this.loadDefault();
+  }
+
+  public async loadDefault(): Promise<void> {
+    const newDefault = this.createDefault();
+    this.default = newDefault;
+
+    if (!this.cacheService.cacheEnabled()) {
+      return;
+    }
+
+    const cacheKey = buildServiceConfigApiRateLimitMaximumKey();
+    const previousHash = await this.cacheService.get(cacheKey);
+    const newHash = this.getConfigHash(newDefault);
+
+    if (previousHash !== newHash) {
+      await this.cacheService.set(cacheKey, newHash);
+
+      this.invalidateCache.invalidateQuery({
+        key: buildMaximumApiRateLimitKey({
```

Comment:
I would not suggest using invalidateQuery with `type: CacheKeyTypeEnum.ENTITY` that is used in buildMaximumApiRateLimitKey, unless i am missing some new implementation that i am not aware of.

invalidateCache.invalidateByKey - responsibility to delete by key
invalidateCache.invalidateQuery - while responsible for deleting all nested SET members that are stored in the redis.

Reply:
Resolved to use the `invalidateByKey` method. Thanks for highlighting.